### DISCUSSION TITLE
Racks overview screen

### DISCUSSION THREAD

**Discussion by Normann33** [netbox-community/netbox#11065] [Q&A]
Hi!

Is it possible to create dashboard, where we can see all our racks and free space in racks? I mean like RackTables

**Comment by candlerb** [comment]
I don't know what RackTables displays.

In the Netbox GUI, under Organization > Racks then you will get a list of racks.  If necessary, click on "Configure Table" to display the "Space" column. (It's also possible to configure the defaults, so that new users get a set of columns that you have chosen).

If you mean you want to write your own dashboard, then you can use the REST API to query the racks in Netbox.  The full list of racks is at `/api/dcim/racks/` and the details of an individual rack at `/api/dcim/racks/<N>/`. However, it seems that neither of these includes the space figure (method `Rack.get_utilization`).  I think this could be raised as a feature request, unless there's another way to obtain this figure via the API that I haven't discovered.

If you mean you want to use some existing third-party dashboard software, say Grafana, then you'll need to get the data out of Netbox and into something that Grafana can query.  For example, you could run a script every hour that copies the list of racks and space utilization into another Postgres table, or into Prometheus Pushgateway.

Getting Grafana to query Netbox's existing tables directly is also theoretically possible, but I wouldn't advise it: firstly the space utilization is not available as a column to read, so you'd have to reimplement this entire logic in SQL; and over time, the schema is subject to change anyway.  (Although having said that, the REST API is also subject to change).

**Reply by Normann33** [reply]
Thanks a lot for your answer!
